lib/propolis/src/enlightenment/hyperv/overlay.rs

```diff
@@ -130,13 +134,45 @@ pub enum OverlayError {
     /// caller.
     #[error("PFN {0:#x} has no overlay of kind {1:?}")]
     NoOverlayOfKind(u64, OverlayKind),
+
+    /// A caller asked to restore the original page contents for a set of
+    /// overlays, but supplied a different number of PFNs to restore than the
+    /// overlay manager is currently tracking.
+    #[error(
+        "imported overlay contents have length {actual_len}, \
+        expected {expected_len}"
+    )]
+    OriginalContentsCountMismatch { actual_len: usize, expected_len: usize },
+
+    /// A caller asked to restore the guest page underlying an invalid PFN.
+    #[error("saved overlay contents PFN {0:#x} is not a valid 52-bit PFN")]
+    InvalidImportPfn(u64),
+
+    /// A caller supplied guest data to use as the original contents of an
+    /// overlaid page, but that data was not page-sized.
+    #[error(
+        "saved overlaid guest page contents are {0} bytes, expected PAGE_SIZE"
+    )]
+    InvalidOverlaidGuestPageLength(usize),
```

Comment:
since this is the complete error message that will bubble up when an import fails, perhaps we ought to include the PFN in _all_ the errors to aid in debugging?

Reply:
Fixed in eaa6f94.